spec/services/find/course_type_answer_determiner_spec.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Find::CourseTypeAnswerDeterminer do
+  describe '#show_undergraduate_courses?' do
+    context 'when the age group is not further education, no degree, and no visa sponsorship required' do
+      let(:params) do
+        {
+          age_group: 'secondary',
+          visa_status: 'false',
+          university_degree_status: 'false'
+        }
+      end
+
+      it 'returns true' do
+        determiner = described_class.new(**params)
+        expect(determiner.show_undergraduate_courses?).to be true
+      end
+    end
+
+    context 'when the age group is further education' do
+      let(:params) do
+        {
+          age_group: 'further_education',
+          visa_status: 'false',
+          university_degree_status: nil
+        }
+      end
+
+      it 'returns false' do
+        determiner = described_class.new(**params)
+        expect(determiner.show_undergraduate_courses?).to be false
+      end
+    end
+
+    context 'when visa status requires sponsorship' do
+      let(:params) do
+        {
+          age_group: 'secondary',
+          visa_status: 'true',
+          university_degree_status: 'false'
+        }
+      end
+
+      it 'returns false' do
+        determiner = described_class.new(**params)
+        expect(determiner.show_undergraduate_courses?).to be false
+      end
+    end
+
+    context 'when there is a university degree' do
+      let(:params) do
+        {
+          age_group: 'secondary',
+          visa_status: 'false',
+          university_degree_status: 'true'
+        }
+      end
+
+      it 'returns false' do
+        determiner = described_class.new(**params)
+        expect(determiner.show_undergraduate_courses?).to be false
+      end
+    end
+
+    context 'when there is not a university degree' do
+      let(:params) do
+        {
+          age_group: 'secondary',
+          visa_status: 'false',
+          university_degree_status: nil
+        }
+      end
+
+      it 'returns false' do
+        determiner = described_class.new(**params)
+        expect(determiner.show_undergraduate_courses?).to be false
+      end
+    end
+  end
+
+  describe '#non_eligible_for_undergraduate_courses?' do
+    context 'when the age group is not further education, visa requires sponsorship, and no degree' do
+      let(:params) do
+        {
+          age_group: 'secondary',
+          visa_status: 'true',
+          university_degree_status: 'false'
+        }
+      end
+
+      it 'returns true' do
+        determiner = described_class.new(**params)
+        expect(determiner.non_eligible_for_undergraduate_courses?).to be true
+      end
+    end
+
+    context 'when the age group is further education' do
+      let(:params) do
+        {
+          age_group: 'further_education',
+          visa_status: 'true',
+          university_degree_status: nil
```

Comment:
Quite a few of these have `university_degree_status` `nil` won't `not_eligible_for_undergraduate_courses?` return false if `university_degree_status` is `nil` I know in reality it has to be true or false for TDA courses.

To test that `not_eligible_for_undergraduate_courses?` is returning false for the right reason, all of the other settings have to be set to the option which will return true.

Could be me just misunderstanding but wanted to check.

Reply:
I see the misunderstanding, the reason that I split into this contexts and set the nil (even if is not needed) this simulates exactly what this class will receive, so I chosen to explicitly about it.

Reply:
Makes sense.